<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>scGraphVerse Case Study: B-cell GRN Reconstruction • scGraphVerse</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="scGraphVerse Case Study: B-cell GRN Reconstruction">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scGraphVerse</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/index.html">Articles</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/francesco-cecere/scGraphVerse/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>scGraphVerse Case Study: B-cell GRN Reconstruction</h1>
                        <h4 data-toc-skip class="author">Francesco
Cecere</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/francesco-cecere/scGraphVerse/blob/HEAD/vignettes/case_study.Rmd" class="external-link"><code>vignettes/case_study.Rmd</code></a></small>
      <div class="d-none name"><code>case_study.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="abstract">Abstract<a class="anchor" aria-label="anchor" href="#abstract"></a>
</h2>
<p>This vignette demonstrates the <strong>scGraphVerse</strong> workflow
on a two-sample B-cell single-cell RNA-seq dataset. We show how to:</p>
<ol style="list-style-type: decimal">
<li>Load and preprocess public PBMC data.</li>
<li>Infer gene regulatory networks with<strong>JRF</strong>.</li>
<li>Build consensus networks and detect communities.</li>
<li>Validate inferred edges using STRINGdb.</li>
</ol>
</div>
<div class="section level2">
<h2 id="dataset-and-preprocessing">1. Dataset and Preprocessing<a class="anchor" aria-label="anchor" href="#dataset-and-preprocessing"></a>
</h2>
<p>We use two public PBMC Seurat objects containing whole-blood samples.
Our goal is to focus on healthy B cells and a common gene set.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">urls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    url1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>        <span class="st">"https://www.dropbox.com/s/r8qwsng79rhp9gf/"</span>,</span>
<span>        <span class="st">"SCA_scRNASEQ_TISSUE_WHOLE_BLOOD.RDS?dl=1"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    url2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>        <span class="st">"https://www.dropbox.com/s/grt5371v5qarn54/"</span>,</span>
<span>        <span class="st">"SCA_scRNASEQ_TISSUE_BLOOD.RDS?dl=1"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># Load Seurat objects</span></span>
<span><span class="va">pbmc1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/download_Atlas.html">download_Atlas</a></span><span class="op">(</span>file_url <span class="op">=</span> <span class="va">urls</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">pbmc2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/download_Atlas.html">download_Atlas</a></span><span class="op">(</span>file_url <span class="op">=</span> <span class="va">urls</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Select top variable genes for B cells, removing ribosomal/mitochondrial</span></span>
<span><span class="va">genes1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selgene.html">selgene</a></span><span class="op">(</span><span class="va">pbmc1</span>,</span>
<span>    top_n <span class="op">=</span> <span class="fl">150</span>,</span>
<span>    cell_type <span class="op">=</span> <span class="st">"B_cell"</span>,</span>
<span>    cell_type_col <span class="op">=</span> <span class="st">"CELL_TYPE"</span>,</span>
<span>    remove_rib <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    remove_mt <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Using Seurat assay 'RNA' slot 'data' (log-normalized).</span></span>
<span><span class="co">#&gt; Subsetted to 197 cells where CELL_TYPE = 'B_cell'.</span></span>
<span><span class="co">#&gt; Removed mitochondrial genes matching '^MT-'.</span></span>
<span><span class="co">#&gt; Removed ribosomal genes matching '^RP[SL]'.</span></span>
<span><span class="co">#&gt; Top 150 genes selected based on mean expression.</span></span>
<span><span class="va">genes2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selgene.html">selgene</a></span><span class="op">(</span><span class="va">pbmc2</span>,</span>
<span>    top_n <span class="op">=</span> <span class="fl">150</span>,</span>
<span>    cell_type <span class="op">=</span> <span class="st">"B_cell"</span>,</span>
<span>    cell_type_col <span class="op">=</span> <span class="st">"CELL_TYPE"</span>,</span>
<span>    remove_rib <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    remove_mt <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Using Seurat assay 'RNA' slot 'data' (log-normalized).</span></span>
<span><span class="co">#&gt; Subsetted to 369 cells where CELL_TYPE = 'B_cell'.</span></span>
<span><span class="co">#&gt; Removed mitochondrial genes matching '^MT-'.</span></span>
<span><span class="co">#&gt; Removed ribosomal genes matching '^RP[SL]'.</span></span>
<span><span class="co">#&gt; Top 150 genes selected based on mean expression.</span></span>
<span></span>
<span><span class="co"># Intersection of gene sets and subset</span></span>
<span><span class="va">common_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">genes1</span>, <span class="va">genes2</span><span class="op">)</span></span>
<span><span class="va">pbmc1_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">pbmc1</span>, features <span class="op">=</span> <span class="va">common_genes</span><span class="op">)</span></span>
<span><span class="va">pbmc2_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">pbmc2</span>, features <span class="op">=</span> <span class="va">common_genes</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Filter to B cells</span></span>
<span><span class="va">b1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">pbmc1_sub</span>, subset <span class="op">=</span> <span class="va">CELL_TYPE</span> <span class="op">==</span> <span class="st">"B_cell"</span><span class="op">)</span></span>
<span><span class="va">b2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">pbmc2_sub</span>, subset <span class="op">=</span> <span class="va">CELL_TYPE</span> <span class="op">==</span> <span class="st">"B_cell"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># List for multi-sample analysis</span></span>
<span><span class="va">bcell_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">b1</span>, <span class="va">b2</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="network-inference">2. Network Inference<a class="anchor" aria-label="anchor" href="#network-inference"></a>
</h2>
<p>We infer GRNs using three algorithms. Replace <code>method</code> to
switch.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Choose method: "GENIE3", "GRNBoost2", or "JRF"</span></span>
<span><span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"GENIE3"</span></span>
<span><span class="va">networks</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/infer_networks.html">infer_networks</a></span><span class="op">(</span></span>
<span>    count_matrices_list <span class="op">=</span> <span class="va">bcell_list</span>,</span>
<span>    method <span class="op">=</span> <span class="va">method</span>,</span>
<span>    nCores <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: The `slot` argument of `GetAssayData()` is deprecated as of SeuratObject 5.0.0.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Please use the `layer` argument instead.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">scGraphVerse</span> package.</span></span>
<span><span class="co">#&gt;   Please report the issue at</span></span>
<span><span class="co">#&gt;   <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/francesco-cecere/scGraphVerse/issues&gt;</span>.</span></span>
<span><span class="co">#&gt; <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;">generated.</span></span></span></code></pre></div>
<div class="section level3">
<h3 id="building-adjacency-matrices">2.1. Building Adjacency Matrices<a class="anchor" aria-label="anchor" href="#building-adjacency-matrices"></a>
</h3>
<p>Convert edge lists to weighted matrices, symmetrize, and apply
threshold.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Weighted adjacency</span></span>
<span><span class="va">wadj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_adjacency.html">generate_adjacency</a></span><span class="op">(</span><span class="va">networks</span><span class="op">)</span></span>
<span><span class="co"># Symmetrize</span></span>
<span><span class="va">swadj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/symmetrize.html">symmetrize</a></span><span class="op">(</span><span class="va">wadj</span>, weight_function <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span></span>
<span><span class="co"># Binary cutoff (top 5%)</span></span>
<span><span class="va">binary_adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cutoff_adjacency.html">cutoff_adjacency</a></span><span class="op">(</span></span>
<span>    count_matrices <span class="op">=</span> <span class="va">bcell_list</span>,</span>
<span>    weighted_adjm_list <span class="op">=</span> <span class="va">swadj</span>,</span>
<span>    n <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    method <span class="op">=</span> <span class="va">method</span>,</span>
<span>    quantile_threshold <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>    nCores <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span>
<span><span class="co"># Plot</span></span>
<span><span class="fu"><a href="../reference/plotg.html">plotg</a></span><span class="op">(</span><span class="va">binary_adj</span><span class="op">)</span></span></code></pre></div>
<p><img src="case_study_files/figure-html/adjacency-1.png" width="576"></p>
</div>
</div>
<div class="section level2">
<h2 id="consensus-and-community-detection">3. Consensus and Community Detection<a class="anchor" aria-label="anchor" href="#consensus-and-community-detection"></a>
</h2>
<p>Aggregate multiple binary networks into a consensus and find network
modules.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Consensus by vote</span></span>
<span><span class="va">consensus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_consensus.html">create_consensus</a></span><span class="op">(</span><span class="va">binary_adj</span>, method <span class="op">=</span> <span class="st">"vote"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plotg.html">plotg</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">consensus</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="case_study_files/figure-html/consensus-1.png" width="576"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/compare_consensus.html">compare_consensus</a></span><span class="op">(</span>consensus_matrix <span class="op">=</span> <span class="va">consensus</span>, false_plot <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Initializing STRINGdb...</span></span>
<span><span class="co">#&gt; Mapping genes to STRING IDs...</span></span>
<span><span class="co">#&gt; Mapped 116 genes to STRING IDs.</span></span>
<span><span class="co">#&gt; Retrieving physical interactions from STRING API...</span></span>
<span><span class="co">#&gt; Found 71 STRING physical interactions.</span></span>
<span><span class="co">#&gt; Adjacency matrices constructed successfully.</span></span></code></pre></div>
<p><img src="case_study_files/figure-html/consensus-2.png" width="576"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Community detection</span></span>
<span><span class="va">communities</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/community_path.html">community_path</a></span><span class="op">(</span><span class="va">consensus</span><span class="op">)</span></span>
<span><span class="co">#&gt; Detecting communities...</span></span></code></pre></div>
<p><img src="case_study_files/figure-html/consensus-3.png" width="576"></p>
<pre><code><span><span class="co">#&gt; Running pathway enrichment...</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; Reading KEGG annotation online: "https://rest.kegg.jp/link/hsa/pathway"...</span></span>
<span><span class="co">#&gt; Reading KEGG annotation online: "https://rest.kegg.jp/list/pathway/hsa"...</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="validation-with-stringdb">4. Validation with STRINGdb<a class="anchor" aria-label="anchor" href="#validation-with-stringdb"></a>
</h2>
<p>Fetch high-confidence interactions from STRING and evaluate true
positives.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">str</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/stringdb_adjacency.html">stringdb_adjacency</a></span><span class="op">(</span></span>
<span>    genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">consensus</span><span class="op">)</span>,</span>
<span>    species <span class="op">=</span> <span class="fl">9606</span>,</span>
<span>    required_score <span class="op">=</span> <span class="fl">900</span>,</span>
<span>    keep_all_genes <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span><span class="op">$</span><span class="va">binary</span></span>
<span><span class="co">#&gt; Initializing STRINGdb...</span></span>
<span><span class="co">#&gt; Mapping genes to STRING IDs...</span></span>
<span><span class="co">#&gt; Mapped 116 genes to STRING IDs.</span></span>
<span><span class="co">#&gt; Retrieving physical interactions from STRING API...</span></span>
<span><span class="co">#&gt; Found 71 STRING physical interactions.</span></span>
<span><span class="co">#&gt; Adjacency matrices constructed successfully.</span></span>
<span><span class="va">ground_truth</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/symmetrize.html">symmetrize</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">str</span><span class="op">)</span>, weight_function <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Edge mining: TP rates</span></span>
<span><span class="va">em</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/edge_mining.html">edge_mining</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">consensus</span><span class="op">)</span>,</span>
<span>    query_edge_types <span class="op">=</span> <span class="st">"TP"</span>,</span>
<span>    ground_truth <span class="op">=</span> <span class="va">ground_truth</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="conclusion">5. Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This case study illustrates how <strong>scGraphVerse</strong> enables
end-to-end GRN reconstruction and validation in single-cell data. Users
can swap inference algorithms, tune thresholds, and incorporate external
prior networks.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=it_IT.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=it_IT.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=it_IT.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=it_IT.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Rome</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] doRNG_1.8.6.2      rngtools_1.5.2     foreach_1.5.2      lubridate_1.9.4   </span></span>
<span><span class="co">#&gt;  [5] forcats_1.0.0      stringr_1.5.1      dplyr_1.1.4        purrr_1.0.4       </span></span>
<span><span class="co">#&gt;  [9] readr_2.1.5        tidyr_1.3.1        tibble_3.2.1       ggplot2_3.5.2     </span></span>
<span><span class="co">#&gt; [13] tidyverse_2.0.0    Seurat_5.3.0       SeuratObject_5.1.0 sp_2.2-0          </span></span>
<span><span class="co">#&gt; [17] scGraphVerse_0.1.0 BiocStyle_2.34.0  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] IRanges_2.40.1              R.methodsS3_1.8.2          </span></span>
<span><span class="co">#&gt;   [3] gld_2.6.7                   goftest_1.2-3              </span></span>
<span><span class="co">#&gt;   [5] Biostrings_2.74.1           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;   [7] ggtangle_0.0.6              spatstat.random_3.3-3      </span></span>
<span><span class="co">#&gt;   [9] perturbR_0.1.3              proxy_0.4-27               </span></span>
<span><span class="co">#&gt;  [11] digest_0.6.37               png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [13] Exact_3.3                   pcaPP_2.0-5                </span></span>
<span><span class="co">#&gt;  [15] ggrepel_0.9.6               deldir_2.0-4               </span></span>
<span><span class="co">#&gt;  [17] parallelly_1.43.0           hdrcde_3.4                 </span></span>
<span><span class="co">#&gt;  [19] MASS_7.3-54                 pkgdown_2.1.2              </span></span>
<span><span class="co">#&gt;  [21] reshape2_1.4.4              httpuv_1.6.16              </span></span>
<span><span class="co">#&gt;  [23] BiocGenerics_0.52.0         qvalue_2.38.0              </span></span>
<span><span class="co">#&gt;  [25] withr_3.0.2                 xfun_0.52                  </span></span>
<span><span class="co">#&gt;  [27] ggfun_0.1.8                 ggpubr_0.6.0               </span></span>
<span><span class="co">#&gt;  [29] survival_3.2-11             memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [31] clusterProfiler_4.14.6      gson_0.1.0                 </span></span>
<span><span class="co">#&gt;  [33] systemfonts_1.2.2           networkD3_0.4.1            </span></span>
<span><span class="co">#&gt;  [35] ragg_1.4.0                  tidytree_0.4.6             </span></span>
<span><span class="co">#&gt;  [37] zoo_1.8-14                  gtools_3.9.5               </span></span>
<span><span class="co">#&gt;  [39] pbapply_1.7-2               R.oo_1.27.0                </span></span>
<span><span class="co">#&gt;  [41] Formula_1.2-6               KEGGREST_1.46.0            </span></span>
<span><span class="co">#&gt;  [43] promises_1.3.2              httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [45] GENIE3_1.28.0               rstatix_0.7.2              </span></span>
<span><span class="co">#&gt;  [47] globals_0.17.0              hash_2.2.6.3               </span></span>
<span><span class="co">#&gt;  [49] fitdistrplus_1.2-2          fmsb_0.7.6                 </span></span>
<span><span class="co">#&gt;  [51] rstudioapi_0.17.1           UCSC.utils_1.2.0           </span></span>
<span><span class="co">#&gt;  [53] miniUI_0.1.2                generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [55] DOSE_4.0.1                  reactome.db_1.89.0         </span></span>
<span><span class="co">#&gt;  [57] curl_6.2.2                  S4Vectors_0.44.0           </span></span>
<span><span class="co">#&gt;  [59] zlibbioc_1.52.0             ggraph_2.2.1               </span></span>
<span><span class="co">#&gt;  [61] polyclip_1.10-7             GenomeInfoDbData_1.2.13    </span></span>
<span><span class="co">#&gt;  [63] SparseArray_1.6.2           xtable_1.8-4               </span></span>
<span><span class="co">#&gt;  [65] desc_1.4.3                  pracma_2.4.4               </span></span>
<span><span class="co">#&gt;  [67] doParallel_1.0.17           evaluate_1.0.3             </span></span>
<span><span class="co">#&gt;  [69] S4Arrays_1.6.0              hms_1.1.3                  </span></span>
<span><span class="co">#&gt;  [71] GenomicRanges_1.58.0        bookdown_0.43              </span></span>
<span><span class="co">#&gt;  [73] irlba_2.3.5.1               colorspace_2.1-1           </span></span>
<span><span class="co">#&gt;  [75] ROCR_1.0-11                 readxl_1.4.5               </span></span>
<span><span class="co">#&gt;  [77] reticulate_1.42.0           spatstat.data_3.1-6        </span></span>
<span><span class="co">#&gt;  [79] magrittr_2.0.3              lmtest_0.9-40              </span></span>
<span><span class="co">#&gt;  [81] later_1.4.2                 viridis_0.6.5              </span></span>
<span><span class="co">#&gt;  [83] ggtree_3.14.0               lattice_0.20-44            </span></span>
<span><span class="co">#&gt;  [85] spatstat.geom_3.3-6         future.apply_1.11.3        </span></span>
<span><span class="co">#&gt;  [87] scattermore_1.2             XML_3.99-0.18              </span></span>
<span><span class="co">#&gt;  [89] cowplot_1.1.3               matrixStats_1.5.0          </span></span>
<span><span class="co">#&gt;  [91] RcppAnnoy_0.0.22            class_7.3-19               </span></span>
<span><span class="co">#&gt;  [93] pillar_1.10.2               nlme_3.1-152               </span></span>
<span><span class="co">#&gt;  [95] iterators_1.0.14            caTools_1.18.3             </span></span>
<span><span class="co">#&gt;  [97] compiler_4.4.2              RSpectra_0.16-2            </span></span>
<span><span class="co">#&gt;  [99] stringi_1.8.7               DescTools_0.99.60          </span></span>
<span><span class="co">#&gt; [101] tensor_1.5                  SummarizedExperiment_1.36.0</span></span>
<span><span class="co">#&gt; [103] plyr_1.8.9                  fda_6.2.0                  </span></span>
<span><span class="co">#&gt; [105] crayon_1.5.3                abind_1.4-8                </span></span>
<span><span class="co">#&gt; [107] gridGraphics_0.5-1          chron_2.3-62               </span></span>
<span><span class="co">#&gt; [109] haven_2.5.4                 graphlayouts_1.2.2         </span></span>
<span><span class="co">#&gt; [111] org.Hs.eg.db_3.20.0         bit_4.6.0                  </span></span>
<span><span class="co">#&gt; [113] rootSolve_1.8.2.4           fastmatch_1.1-6            </span></span>
<span><span class="co">#&gt; [115] codetools_0.2-18            textshaping_1.0.0          </span></span>
<span><span class="co">#&gt; [117] bslib_0.9.0                 e1071_1.7-16               </span></span>
<span><span class="co">#&gt; [119] lmom_3.2                    fds_1.8                    </span></span>
<span><span class="co">#&gt; [121] plotly_4.10.4               mime_0.13                  </span></span>
<span><span class="co">#&gt; [123] multinet_4.2.2              splines_4.4.2              </span></span>
<span><span class="co">#&gt; [125] Rcpp_1.0.14                 fastDummies_1.7.5          </span></span>
<span><span class="co">#&gt; [127] cellranger_1.1.0            datastructures_0.2.9       </span></span>
<span><span class="co">#&gt; [129] knitr_1.50                  blob_1.2.4                 </span></span>
<span><span class="co">#&gt; [131] robin_2.1.0                 fs_1.6.6                   </span></span>
<span><span class="co">#&gt; [133] listenv_0.9.1               expm_1.0-0                 </span></span>
<span><span class="co">#&gt; [135] ggsignif_0.6.4              ggplotify_0.1.2            </span></span>
<span><span class="co">#&gt; [137] sqldf_0.4-11                Matrix_1.7-3               </span></span>
<span><span class="co">#&gt; [139] tzdb_0.5.0                  tweenr_2.0.3               </span></span>
<span><span class="co">#&gt; [141] pkgconfig_2.0.3             tools_4.4.2                </span></span>
<span><span class="co">#&gt; [143] cachem_1.1.0                RSQLite_2.3.9              </span></span>
<span><span class="co">#&gt; [145] viridisLite_0.4.2           DBI_1.2.3                  </span></span>
<span><span class="co">#&gt; [147] distributions3_0.2.2        graphite_1.52.0            </span></span>
<span><span class="co">#&gt; [149] fastmap_1.2.0               rmarkdown_2.29             </span></span>
<span><span class="co">#&gt; [151] scales_1.3.0                grid_4.4.2                 </span></span>
<span><span class="co">#&gt; [153] ica_1.0-3                   broom_1.0.8                </span></span>
<span><span class="co">#&gt; [155] sass_0.4.10                 patchwork_1.3.0            </span></span>
<span><span class="co">#&gt; [157] INetTool_0.1.0              BiocManager_1.30.25        </span></span>
<span><span class="co">#&gt; [159] dotCall64_1.2               graph_1.84.1               </span></span>
<span><span class="co">#&gt; [161] carData_3.0-5               RANN_2.6.2                 </span></span>
<span><span class="co">#&gt; [163] farver_2.1.2                tidygraph_1.3.1            </span></span>
<span><span class="co">#&gt; [165] gsubfn_0.7                  yaml_2.3.10                </span></span>
<span><span class="co">#&gt; [167] deSolve_1.40                MatrixGenerics_1.18.1      </span></span>
<span><span class="co">#&gt; [169] cli_3.6.5                   stats4_4.4.2               </span></span>
<span><span class="co">#&gt; [171] lifecycle_1.0.4             askpass_1.2.1              </span></span>
<span><span class="co">#&gt; [173] uwot_0.2.3                  Biobase_2.66.0             </span></span>
<span><span class="co">#&gt; [175] rainbow_3.8                 mvtnorm_1.3-3              </span></span>
<span><span class="co">#&gt; [177] backports_1.5.0             BiocParallel_1.40.2        </span></span>
<span><span class="co">#&gt; [179] timechange_0.3.0            gtable_0.3.6               </span></span>
<span><span class="co">#&gt; [181] ggridges_0.5.6              progressr_0.15.1           </span></span>
<span><span class="co">#&gt; [183] parallel_4.4.2              pROC_1.18.5                </span></span>
<span><span class="co">#&gt; [185] ape_5.8-1                   jsonlite_2.0.0             </span></span>
<span><span class="co">#&gt; [187] RcppHNSW_0.6.0              bitops_1.0-9               </span></span>
<span><span class="co">#&gt; [189] bit64_4.6.0-1               Rtsne_0.17                 </span></span>
<span><span class="co">#&gt; [191] yulab.utils_0.2.0           ReactomePA_1.50.0          </span></span>
<span><span class="co">#&gt; [193] spatstat.utils_3.1-3        proto_1.0.0                </span></span>
<span><span class="co">#&gt; [195] jquerylib_0.1.4             GOSemSim_2.32.0            </span></span>
<span><span class="co">#&gt; [197] spatstat.univar_3.1-2       R.utils_2.13.0             </span></span>
<span><span class="co">#&gt; [199] lazyeval_0.2.2              shiny_1.10.0               </span></span>
<span><span class="co">#&gt; [201] htmltools_0.5.8.1           enrichplot_1.26.6          </span></span>
<span><span class="co">#&gt; [203] GO.db_3.20.0                sctransform_0.4.1          </span></span>
<span><span class="co">#&gt; [205] data.tree_1.1.0             rappdirs_0.3.3             </span></span>
<span><span class="co">#&gt; [207] glue_1.8.0                  STRINGdb_2.18.0            </span></span>
<span><span class="co">#&gt; [209] spam_2.11-1                 XVector_0.46.0             </span></span>
<span><span class="co">#&gt; [211] qpdf_1.3.5                  RCurl_1.98-1.17            </span></span>
<span><span class="co">#&gt; [213] treeio_1.30.0               mclust_6.1.1               </span></span>
<span><span class="co">#&gt; [215] ks_1.14.3                   gridExtra_2.3              </span></span>
<span><span class="co">#&gt; [217] boot_1.3-28                 igraph_2.1.4               </span></span>
<span><span class="co">#&gt; [219] R6_2.6.1                    fdatest_2.1.1              </span></span>
<span><span class="co">#&gt; [221] SingleCellExperiment_1.28.1 gplots_3.2.0               </span></span>
<span><span class="co">#&gt; [223] labeling_0.4.3              cluster_2.1.2              </span></span>
<span><span class="co">#&gt; [225] aplot_0.2.5                 GenomeInfoDb_1.42.3        </span></span>
<span><span class="co">#&gt; [227] DelayedArray_0.32.0         tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt; [229] plotrix_3.8-4               ggforce_0.4.2              </span></span>
<span><span class="co">#&gt; [231] car_3.1-3                   AnnotationDbi_1.68.0       </span></span>
<span><span class="co">#&gt; [233] future_1.40.0               munsell_0.5.1              </span></span>
<span><span class="co">#&gt; [235] KernSmooth_2.23-20          data.table_1.17.0          </span></span>
<span><span class="co">#&gt; [237] htmlwidgets_1.6.4           fgsea_1.32.4               </span></span>
<span><span class="co">#&gt; [239] RColorBrewer_1.1-3          rlang_1.1.6                </span></span>
<span><span class="co">#&gt; [241] spatstat.sparse_3.1-0       spatstat.explore_3.4-2     </span></span>
<span><span class="co">#&gt; [243] rentrez_1.2.3</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Francesco Cecere.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
