<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started with scGraphVerse • scGraphVerse</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="48x48" href="../favicon-48x48.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started with scGraphVerse">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scGraphVerse</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/simulation_study.html">Getting Started with scGraphVerse</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/francesco-cecere/scGraphVerse/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Started with scGraphVerse</h1>
                        <h4 data-toc-skip class="author">Francesco
Cecere</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/francesco-cecere/scGraphVerse/blob/HEAD/vignettes/simulation_study.Rmd" class="external-link"><code>vignettes/simulation_study.Rmd</code></a></small>
      <div class="hidden name"><code>simulation_study.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="bioconductor-compliance">Bioconductor compliance<a class="anchor" aria-label="anchor" href="#bioconductor-compliance"></a>
</h2>
<p><strong>scGraphVerse</strong> is distributed through Bioconductor; it
adheres to the <a href="https://bioconductor.org/developers/package-guidelines/" class="external-link">Bioconductor
package guidelines</a> for documentation, versioning, and testing.
Please make sure to run <code><a href="https://rdrr.io/pkg/BiocCheck/man/BiocCheck.html" class="external-link">BiocCheck::BiocCheck()</a></code> before
submission, include unit tests under <code>tests/</code>, and follow the
code‐of‐conduct.</p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Single‐cell RNA‐seq has opened new doors for reconstructing
cell‐type‐specific gene regulatory networks (GRNs).
<em>scGraphVerse</em> provides a streamlined workflow to:</p>
<ol style="list-style-type: decimal">
<li>Simulate single‐cell count data (including zero‐inflation) via the
ZILGM framework[^ZILGM]<br>
</li>
<li>Infer GRNs with multiple algorithms:
<ul>
<li>
<strong>GENIE3</strong>[^GENIE3]<br>
</li>
<li>
<strong>GRNBoost2</strong>[^GRNBoost2]<br>
</li>
<li>
<strong>JRF</strong>[^JRF]<br>
</li>
</ul>
</li>
<li>Evaluate performance against a ground‐truth network<br>
</li>
<li>Build consensus networks and explore network communities</li>
</ol>
<p>Throughout this vignette we’ll highlight the core functions and point
you to the original method papers.</p>
</div>
<div class="section level2">
<h2 id="load-and-preprocess-a-public-dataset">Load and Pre‐process a Public Dataset<a class="anchor" aria-label="anchor" href="#load-and-preprocess-a-public-dataset"></a>
</h2>
<p>We start by downloading a PBMC dataset from the CellxGene atlas,
subsetting to healthy CD4⁺ T cells, and applying standard QC filters
(mitochondrial, ribosomal, feature and count thresholds).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># options(timeout = 2000)</span></span>
<span></span>
<span><span class="co"># 1. Download the Seurat object</span></span>
<span><span class="va">dataset_url</span> <span class="op">&lt;-</span> <span class="st">"https://www.dropbox.com/s/r8qwsng79rhp9gf/SCA_scRNASEQ_TISSUE_WHOLE_BLOOD.RDS?dl=1"</span></span>
<span><span class="va">sc_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/download_Atlas.html">download_Atlas</a></span><span class="op">(</span>file_url <span class="op">=</span> <span class="va">dataset_url</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="define-a-groundtruth-regulatory-network">Define a “Ground‐Truth” Regulatory Network<a class="anchor" aria-label="anchor" href="#define-a-groundtruth-regulatory-network"></a>
</h2>
<p>We use high‐confidence STRINGdb edges as our ground truth. You can
substitute any known adjacency matrix here.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Select top variable genes</span></span>
<span><span class="va">pathgenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/selgene.html">selgene</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">sc_object</span>,</span>
<span>  top_n <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  cell_type <span class="op">=</span> <span class="st">"T_cells"</span>,</span>
<span>  cell_type_col <span class="op">=</span> <span class="st">"CELL_TYPE"</span>,</span>
<span>  remove_rib <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  remove_mt <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Using Seurat assay 'RNA' slot 'data' (log-normalized).</span></span>
<span><span class="co">#&gt; Subsetted to 923 cells where CELL_TYPE = 'T_cells'.</span></span>
<span><span class="co">#&gt; Removed mitochondrial genes matching '^MT-'.</span></span>
<span><span class="co">#&gt; Removed ribosomal genes matching '^RP[SL]'.</span></span>
<span><span class="co">#&gt; Top 500 genes selected based on mean expression.</span></span>
<span></span>
<span><span class="co"># 2. Query STRINGdb for high‐confidence edges</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/stringdb_adjacency.html">stringdb_adjacency</a></span><span class="op">(</span></span>
<span>  genes          <span class="op">=</span> <span class="va">pathgenes</span>,</span>
<span>  species        <span class="op">=</span> <span class="fl">9606</span>,</span>
<span>  required_score <span class="op">=</span> <span class="fl">900</span>,</span>
<span>  keep_all_genes <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Initializing STRINGdb...</span></span>
<span><span class="co">#&gt; Mapping genes to STRING IDs...</span></span>
<span><span class="co">#&gt; Mapped 500 genes to STRING IDs.</span></span>
<span><span class="co">#&gt; Retrieving **physical** interactions from STRING API...</span></span>
<span><span class="co">#&gt; Found 377 STRING physical interactions.</span></span>
<span><span class="co">#&gt; Adjacency matrices constructed successfully.</span></span>
<span></span>
<span><span class="va">wadj</span> <span class="op">&lt;-</span> <span class="va">result</span><span class="op">$</span><span class="va">weighted</span></span>
<span><span class="va">adj</span> <span class="op">&lt;-</span> <span class="va">result</span><span class="op">$</span><span class="va">binary</span></span>
<span></span>
<span><span class="co"># 3. Symmetrize and sort</span></span>
<span><span class="va">common</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">adj</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">adj</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">adj</span> <span class="op">&lt;-</span> <span class="va">adj</span><span class="op">[</span><span class="va">common</span>, <span class="va">common</span><span class="op">]</span></span>
<span><span class="va">adj</span> <span class="op">&lt;-</span> <span class="va">adj</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">adj</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">adj</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># 4. Visualize</span></span>
<span><span class="va">gtruth</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/graph_from_adjacency_matrix.html" class="external-link">graph_from_adjacency_matrix</a></span><span class="op">(</span><span class="va">adj</span>, mode <span class="op">=</span> <span class="st">"undirected"</span><span class="op">)</span></span>
<span><span class="fu">ggraph</span><span class="fu">::</span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/ggraph.html" class="external-link">ggraph</a></span><span class="op">(</span><span class="va">gtruth</span>, layout <span class="op">=</span> <span class="st">"fr"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggraph</span><span class="fu">::</span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_edge_link.html" class="external-link">geom_edge_link</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"gray"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggraph</span><span class="fu">::</span><span class="fu"><a href="https://ggraph.data-imaginist.com/reference/geom_node_point.html" class="external-link">geom_node_point</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"steelblue"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>      <span class="st">"Ground Truth: "</span>, <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/gorder.html" class="external-link">vcount</a></span><span class="op">(</span><span class="va">gtruth</span><span class="op">)</span>, <span class="st">" nodes, "</span>,</span>
<span>      <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://r.igraph.org/reference/gsize.html" class="external-link">ecount</a></span><span class="op">(</span><span class="va">gtruth</span><span class="op">)</span>, <span class="st">" edges"</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">+</span> <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="simulation_study_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="simulate-zeroinflated-count-data">Simulate Zero‐Inflated Count Data<a class="anchor" aria-label="anchor" href="#simulate-zeroinflated-count-data"></a>
</h2>
<p>We leverage a zero‐inflated network‐aware simulator based on the
ZILGM framework1, which retains the ground‐truth dependencies in the
counts.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nodes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">adj</span><span class="op">)</span></span>
<span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">sigma</span> <span class="op">&lt;-</span> <span class="fl">0.2</span></span>
<span></span>
<span><span class="va">count_matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/zinb_simdata.html">zinb_simdata</a></span><span class="op">(</span></span>
<span>  n <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  p <span class="op">=</span> <span class="va">nodes</span>,</span>
<span>  B <span class="op">=</span> <span class="va">adj</span>,</span>
<span>  mu_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">4</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">7</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  mu_noise <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>  theta <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0.7</span>, <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>  pi <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.2</span>, <span class="fl">0.2</span><span class="op">)</span>,</span>
<span>  kmat <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  depth_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="op">(</span><span class="fl">1</span> <span class="op">-</span> <span class="va">sigma</span><span class="op">)</span> <span class="op">*</span> <span class="va">nodes</span> <span class="op">*</span> <span class="va">b</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="op">(</span><span class="fl">1</span> <span class="op">+</span> <span class="va">sigma</span><span class="op">)</span> <span class="op">*</span> <span class="va">nodes</span> <span class="op">*</span> <span class="va">b</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Transpose each to cells × genes</span></span>
<span><span class="va">count_matrices</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">count_matrices</span>, <span class="va">t</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="infer-networks-with-multiple-algorithms">Infer Networks with Multiple Algorithms<a class="anchor" aria-label="anchor" href="#infer-networks-with-multiple-algorithms"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#networks_late &lt;- infer_networks(</span></span>
<span><span class="co">#  count_matrices,</span></span>
<span><span class="co">#  method = "GENIE3",</span></span>
<span><span class="co">#  nCores = 15</span></span>
<span><span class="co">#)</span></span>
<span></span>
<span><span class="co">#list_wadj &lt;- generate_adjacency(networks_late)</span></span>
<span><span class="co">#list_swadj &lt;- symmetrize(list_wadj, weight_function = "mean")</span></span>
<span></span>
<span><span class="co">#binary_adjm_list &lt;- cutoff_adjacency(</span></span>
<span><span class="co">#  count_matrices = count_matrices,</span></span>
<span><span class="co">#  weighted_adjm_list = list_swadj,</span></span>
<span><span class="co">#  n = 2,</span></span>
<span><span class="co">#  method = "GENIE3",</span></span>
<span><span class="co">#  quantile_threshold = 0.95,</span></span>
<span><span class="co">#  nCores = 15,</span></span>
<span><span class="co">#  debug = TRUE</span></span>
<span><span class="co">#)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#earlym &lt;- list(earlyj(count_matrices))</span></span>
<span><span class="co">#networks_early &lt;- infer_networks(</span></span>
<span><span class="co">#  earlym,</span></span>
<span><span class="co">#  method = "GENIE3",</span></span>
<span><span class="co">#  nCores = 15</span></span>
<span><span class="co">#)</span></span>
<span></span>
<span><span class="co">#list_wadj &lt;- generate_adjacency(networks_early)</span></span>
<span><span class="co">#list_swadj &lt;- symmetrize(list_wadj, weight_function = "mean")</span></span>
<span></span>
<span><span class="co">#binary_adjm_list_early &lt;- cutoff_adjacency(</span></span>
<span><span class="co">#  count_matrices = earlym,</span></span>
<span><span class="co">#  weighted_adjm_list = list_swadj,</span></span>
<span><span class="co">#  n = 2,</span></span>
<span><span class="co">#  method = "GENIE3",</span></span>
<span><span class="co">#  quantile_threshold = 0.95,</span></span>
<span><span class="co">#  nCores = 15,</span></span>
<span><span class="co">#  debug = TRUE</span></span>
<span><span class="co">#)</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">networks_joint</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/infer_networks.html">infer_networks</a></span><span class="op">(</span></span>
<span>  count_matrices_list <span class="op">=</span> <span class="va">count_matrices</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"JRF"</span>,</span>
<span>  nCores <span class="op">=</span> <span class="fl">15</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">list_wadjj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_adjacency.html">generate_adjacency</a></span><span class="op">(</span><span class="va">networks_joint</span><span class="op">)</span></span>
<span><span class="va">list_swadjj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/symmetrize.html">symmetrize</a></span><span class="op">(</span><span class="va">list_wadjj</span>, weight_function <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">binary_adjm_listj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cutoff_adjacency.html">cutoff_adjacency</a></span><span class="op">(</span></span>
<span>  count_matrices <span class="op">=</span> <span class="va">count_matrices</span>,</span>
<span>  weighted_adjm_list <span class="op">=</span> <span class="va">list_swadjj</span>,</span>
<span>  n <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"JRF"</span>,</span>
<span>  quantile_threshold <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>  nCores <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  debug <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [Method: JRF] Matrix 1 → Cutoff = 0.55335</span></span>
<span><span class="co">#&gt; [Method: JRF] Matrix 2 → Cutoff = 0.58835</span></span>
<span><span class="co">#&gt; [Method: JRF] Matrix 3 → Cutoff = 0.58740</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">joint_auc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotROC.html">plotROC</a></span><span class="op">(</span><span class="va">list_swadjj</span>, <span class="va">adj</span>, plot_title <span class="op">=</span> <span class="st">"ROC curve - JRF Joint Integration"</span>, is_binary <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">joint_auc</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<p><img src="simulation_study_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scores.joint</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pscores.html">pscores</a></span><span class="op">(</span><span class="va">adj</span>, <span class="va">binary_adjm_listj</span><span class="op">)</span></span></code></pre></div>
<p><img src="simulation_study_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotg.html">plotg</a></span><span class="op">(</span><span class="va">binary_adjm_listj</span><span class="op">)</span></span></code></pre></div>
<p><img src="simulation_study_files/figure-html/unnamed-chunk-8-2.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">consesusm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_consensus.html">create_consensus</a></span><span class="op">(</span><span class="va">binary_adjm_listj</span>, method<span class="op">=</span><span class="st">"vote"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plotg.html">plotg</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">consesusm</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="simulation_study_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/compare_consensus.html">compare_consensus</a></span><span class="op">(</span>consensus_matrix <span class="op">=</span> <span class="va">consesusm</span>, reference_matrix <span class="op">=</span> <span class="va">adj</span>, false_plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="simulation_study_files/figure-html/unnamed-chunk-9-2.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">scores.vote</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pscores.html">pscores</a></span><span class="op">(</span><span class="va">adj</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">consesusm</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="simulation_study_files/figure-html/unnamed-chunk-9-3.png" width="700"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adj_comm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/community_path.html">community_path</a></span><span class="op">(</span><span class="va">adj</span><span class="op">)</span></span>
<span><span class="co">#&gt; Detecting communities...</span></span></code></pre></div>
<p><img src="simulation_study_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Running pathway enrichment...</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; Reading KEGG annotation online: "https://rest.kegg.jp/link/hsa/pathway"...</span></span>
<span><span class="co">#&gt; Reading KEGG annotation online: "https://rest.kegg.jp/list/pathway/hsa"...</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="va">comm_consesusm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/community_path.html">community_path</a></span><span class="op">(</span><span class="va">consesusm</span><span class="op">)</span></span>
<span><span class="co">#&gt; Detecting communities...</span></span></code></pre>
<p><img src="simulation_study_files/figure-html/unnamed-chunk-10-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; Running pathway enrichment...</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span>
<span><span class="va">topscore</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/community_similarity.html">community_similarity</a></span><span class="op">(</span><span class="va">adj_comm</span>,<span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">comm_consesusm</span><span class="op">)</span><span class="op">)</span> </span></code></pre>
<p><img src="simulation_study_files/figure-html/unnamed-chunk-10-3.png" width="700"><img src="simulation_study_files/figure-html/unnamed-chunk-10-4.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/edge_mining.html">edge_mining</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">consesusm</span><span class="op">)</span>, <span class="va">adj</span>, query_edge_types <span class="op">=</span> <span class="st">"TP"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=it_IT.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=it_IT.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=it_IT.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=it_IT.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Rome</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] lubridate_1.9.4    forcats_1.0.0      stringr_1.5.1      dplyr_1.1.4       </span></span>
<span><span class="co">#&gt;  [5] purrr_1.0.4        readr_2.1.5        tidyr_1.3.1        tibble_3.2.1      </span></span>
<span><span class="co">#&gt;  [9] ggplot2_3.5.2      tidyverse_2.0.0    Seurat_5.3.0       SeuratObject_5.1.0</span></span>
<span><span class="co">#&gt; [13] sp_2.2-0           scGraphVerse_0.1.0 BiocStyle_2.34.0  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] IRanges_2.40.1              R.methodsS3_1.8.2          </span></span>
<span><span class="co">#&gt;   [3] gld_2.6.7                   goftest_1.2-3              </span></span>
<span><span class="co">#&gt;   [5] Biostrings_2.74.1           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;   [7] ggtangle_0.0.6              spatstat.random_3.3-3      </span></span>
<span><span class="co">#&gt;   [9] perturbR_0.1.3              proxy_0.4-27               </span></span>
<span><span class="co">#&gt;  [11] digest_0.6.37               png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [13] Exact_3.3                   pcaPP_2.0-5                </span></span>
<span><span class="co">#&gt;  [15] ggrepel_0.9.6               deldir_2.0-4               </span></span>
<span><span class="co">#&gt;  [17] parallelly_1.43.0           hdrcde_3.4                 </span></span>
<span><span class="co">#&gt;  [19] MASS_7.3-54                 pkgdown_2.1.2              </span></span>
<span><span class="co">#&gt;  [21] reshape2_1.4.4              httpuv_1.6.16              </span></span>
<span><span class="co">#&gt;  [23] foreach_1.5.2               BiocGenerics_0.52.0        </span></span>
<span><span class="co">#&gt;  [25] qvalue_2.38.0               withr_3.0.2                </span></span>
<span><span class="co">#&gt;  [27] xfun_0.52                   ggfun_0.1.8                </span></span>
<span><span class="co">#&gt;  [29] ggpubr_0.6.0                survival_3.2-11            </span></span>
<span><span class="co">#&gt;  [31] memoise_2.0.1               clusterProfiler_4.14.6     </span></span>
<span><span class="co">#&gt;  [33] gson_0.1.0                  systemfonts_1.2.2          </span></span>
<span><span class="co">#&gt;  [35] networkD3_0.4.1             ragg_1.4.0                 </span></span>
<span><span class="co">#&gt;  [37] tidytree_0.4.6              zoo_1.8-14                 </span></span>
<span><span class="co">#&gt;  [39] gtools_3.9.5                pbapply_1.7-2              </span></span>
<span><span class="co">#&gt;  [41] R.oo_1.27.0                 Formula_1.2-6              </span></span>
<span><span class="co">#&gt;  [43] KEGGREST_1.46.0             promises_1.3.2             </span></span>
<span><span class="co">#&gt;  [45] httr_1.4.7                  rstatix_0.7.2              </span></span>
<span><span class="co">#&gt;  [47] globals_0.17.0              hash_2.2.6.3               </span></span>
<span><span class="co">#&gt;  [49] fitdistrplus_1.2-2          fmsb_0.7.6                 </span></span>
<span><span class="co">#&gt;  [51] rstudioapi_0.17.1           UCSC.utils_1.2.0           </span></span>
<span><span class="co">#&gt;  [53] miniUI_0.1.2                generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [55] DOSE_4.0.1                  reactome.db_1.89.0         </span></span>
<span><span class="co">#&gt;  [57] curl_6.2.2                  S4Vectors_0.44.0           </span></span>
<span><span class="co">#&gt;  [59] zlibbioc_1.52.0             ggraph_2.2.1               </span></span>
<span><span class="co">#&gt;  [61] polyclip_1.10-7             GenomeInfoDbData_1.2.13    </span></span>
<span><span class="co">#&gt;  [63] SparseArray_1.6.2           xtable_1.8-4               </span></span>
<span><span class="co">#&gt;  [65] desc_1.4.3                  pracma_2.4.4               </span></span>
<span><span class="co">#&gt;  [67] doParallel_1.0.17           JRF_0.1-4                  </span></span>
<span><span class="co">#&gt;  [69] evaluate_1.0.3              S4Arrays_1.6.0             </span></span>
<span><span class="co">#&gt;  [71] hms_1.1.3                   GenomicRanges_1.58.0       </span></span>
<span><span class="co">#&gt;  [73] bookdown_0.43               irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt;  [75] colorspace_2.1-1            ROCR_1.0-11                </span></span>
<span><span class="co">#&gt;  [77] readxl_1.4.5                reticulate_1.42.0          </span></span>
<span><span class="co">#&gt;  [79] spatstat.data_3.1-6         magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;  [81] lmtest_0.9-40               later_1.4.2                </span></span>
<span><span class="co">#&gt;  [83] viridis_0.6.5               ggtree_3.14.0              </span></span>
<span><span class="co">#&gt;  [85] lattice_0.20-44             spatstat.geom_3.3-6        </span></span>
<span><span class="co">#&gt;  [87] future.apply_1.11.3         scattermore_1.2            </span></span>
<span><span class="co">#&gt;  [89] XML_3.99-0.18               cowplot_1.1.3              </span></span>
<span><span class="co">#&gt;  [91] matrixStats_1.5.0           RcppAnnoy_0.0.22           </span></span>
<span><span class="co">#&gt;  [93] class_7.3-19                pillar_1.10.2              </span></span>
<span><span class="co">#&gt;  [95] nlme_3.1-152                iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [97] caTools_1.18.3              compiler_4.4.2             </span></span>
<span><span class="co">#&gt;  [99] RSpectra_0.16-2             stringi_1.8.7              </span></span>
<span><span class="co">#&gt; [101] DescTools_0.99.60           tensor_1.5                 </span></span>
<span><span class="co">#&gt; [103] SummarizedExperiment_1.36.0 plyr_1.8.9                 </span></span>
<span><span class="co">#&gt; [105] fda_6.2.0                   crayon_1.5.3               </span></span>
<span><span class="co">#&gt; [107] abind_1.4-8                 gridGraphics_0.5-1         </span></span>
<span><span class="co">#&gt; [109] chron_2.3-62                haven_2.5.4                </span></span>
<span><span class="co">#&gt; [111] graphlayouts_1.2.2          org.Hs.eg.db_3.20.0        </span></span>
<span><span class="co">#&gt; [113] bit_4.6.0                   rootSolve_1.8.2.4          </span></span>
<span><span class="co">#&gt; [115] fastmatch_1.1-6             codetools_0.2-18           </span></span>
<span><span class="co">#&gt; [117] textshaping_1.0.0           bslib_0.9.0                </span></span>
<span><span class="co">#&gt; [119] e1071_1.7-16                lmom_3.2                   </span></span>
<span><span class="co">#&gt; [121] fds_1.8                     plotly_4.10.4              </span></span>
<span><span class="co">#&gt; [123] mime_0.13                   multinet_4.2.2             </span></span>
<span><span class="co">#&gt; [125] splines_4.4.2               Rcpp_1.0.14                </span></span>
<span><span class="co">#&gt; [127] fastDummies_1.7.5           cellranger_1.1.0           </span></span>
<span><span class="co">#&gt; [129] datastructures_0.2.9        knitr_1.50                 </span></span>
<span><span class="co">#&gt; [131] blob_1.2.4                  robin_2.1.0                </span></span>
<span><span class="co">#&gt; [133] fs_1.6.6                    listenv_0.9.1              </span></span>
<span><span class="co">#&gt; [135] expm_1.0-0                  ggsignif_0.6.4             </span></span>
<span><span class="co">#&gt; [137] ggplotify_0.1.2             sqldf_0.4-11               </span></span>
<span><span class="co">#&gt; [139] Matrix_1.7-3                tzdb_0.5.0                 </span></span>
<span><span class="co">#&gt; [141] tweenr_2.0.3                pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [143] tools_4.4.2                 cachem_1.1.0               </span></span>
<span><span class="co">#&gt; [145] RSQLite_2.3.9               viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [147] DBI_1.2.3                   distributions3_0.2.2       </span></span>
<span><span class="co">#&gt; [149] graphite_1.52.0             fastmap_1.2.0              </span></span>
<span><span class="co">#&gt; [151] rmarkdown_2.29              scales_1.3.0               </span></span>
<span><span class="co">#&gt; [153] grid_4.4.2                  ica_1.0-3                  </span></span>
<span><span class="co">#&gt; [155] broom_1.0.8                 sass_0.4.10                </span></span>
<span><span class="co">#&gt; [157] patchwork_1.3.0             INetTool_0.1.0             </span></span>
<span><span class="co">#&gt; [159] BiocManager_1.30.25         dotCall64_1.2              </span></span>
<span><span class="co">#&gt; [161] graph_1.84.1                carData_3.0-5              </span></span>
<span><span class="co">#&gt; [163] RANN_2.6.2                  farver_2.1.2               </span></span>
<span><span class="co">#&gt; [165] tidygraph_1.3.1             gsubfn_0.7                 </span></span>
<span><span class="co">#&gt; [167] yaml_2.3.10                 deSolve_1.40               </span></span>
<span><span class="co">#&gt; [169] MatrixGenerics_1.18.1       cli_3.6.5                  </span></span>
<span><span class="co">#&gt; [171] stats4_4.4.2                lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt; [173] askpass_1.2.1               uwot_0.2.3                 </span></span>
<span><span class="co">#&gt; [175] Biobase_2.66.0              rainbow_3.8                </span></span>
<span><span class="co">#&gt; [177] mvtnorm_1.3-3               backports_1.5.0            </span></span>
<span><span class="co">#&gt; [179] BiocParallel_1.40.2         timechange_0.3.0           </span></span>
<span><span class="co">#&gt; [181] gtable_0.3.6                ggridges_0.5.6             </span></span>
<span><span class="co">#&gt; [183] progressr_0.15.1            parallel_4.4.2             </span></span>
<span><span class="co">#&gt; [185] pROC_1.18.5                 ape_5.8-1                  </span></span>
<span><span class="co">#&gt; [187] jsonlite_2.0.0              RcppHNSW_0.6.0             </span></span>
<span><span class="co">#&gt; [189] bitops_1.0-9                bit64_4.6.0-1              </span></span>
<span><span class="co">#&gt; [191] Rtsne_0.17                  yulab.utils_0.2.0          </span></span>
<span><span class="co">#&gt; [193] ReactomePA_1.50.0           spatstat.utils_3.1-3       </span></span>
<span><span class="co">#&gt; [195] proto_1.0.0                 jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [197] GOSemSim_2.32.0             spatstat.univar_3.1-2      </span></span>
<span><span class="co">#&gt; [199] R.utils_2.13.0              lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [201] shiny_1.10.0                htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt; [203] enrichplot_1.26.6           GO.db_3.20.0               </span></span>
<span><span class="co">#&gt; [205] sctransform_0.4.1           data.tree_1.1.0            </span></span>
<span><span class="co">#&gt; [207] rappdirs_0.3.3              glue_1.8.0                 </span></span>
<span><span class="co">#&gt; [209] STRINGdb_2.18.0             spam_2.11-1                </span></span>
<span><span class="co">#&gt; [211] XVector_0.46.0              qpdf_1.3.5                 </span></span>
<span><span class="co">#&gt; [213] RCurl_1.98-1.17             treeio_1.30.0              </span></span>
<span><span class="co">#&gt; [215] mclust_6.1.1                ks_1.14.3                  </span></span>
<span><span class="co">#&gt; [217] gridExtra_2.3               boot_1.3-28                </span></span>
<span><span class="co">#&gt; [219] igraph_2.1.4                R6_2.6.1                   </span></span>
<span><span class="co">#&gt; [221] fdatest_2.1.1               SingleCellExperiment_1.28.1</span></span>
<span><span class="co">#&gt; [223] gplots_3.2.0                labeling_0.4.3             </span></span>
<span><span class="co">#&gt; [225] cluster_2.1.2               aplot_0.2.5                </span></span>
<span><span class="co">#&gt; [227] GenomeInfoDb_1.42.3         DelayedArray_0.32.0        </span></span>
<span><span class="co">#&gt; [229] tidyselect_1.2.1            plotrix_3.8-4              </span></span>
<span><span class="co">#&gt; [231] ggforce_0.4.2               car_3.1-3                  </span></span>
<span><span class="co">#&gt; [233] AnnotationDbi_1.68.0        future_1.40.0              </span></span>
<span><span class="co">#&gt; [235] munsell_0.5.1               KernSmooth_2.23-20         </span></span>
<span><span class="co">#&gt; [237] data.table_1.17.0           htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt; [239] fgsea_1.32.4                RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt; [241] rlang_1.1.6                 spatstat.sparse_3.1-0      </span></span>
<span><span class="co">#&gt; [243] spatstat.explore_3.4-2      rentrez_1.2.3</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Francesco Cecere.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer>
</div>






  </body>
</html>
